ludwig/hyperopt/execution.py

```diff
@@ -741,21 +740,13 @@ def execute(
         metric = "metric_score"
         use_gpu = bool(self._gpu_resources_per_trial_non_none)
 
-        # if random seed not set, use Ludwig seed
-        self.search_algorithm.check_for_random_seed(random_seed)
-        if self.search_algorithm.search_alg_dict is not None:
-            if TYPE not in self.search_algorithm.search_alg_dict:
-                candiate_search_algs = [search_alg for search_alg in SEARCH_ALG_IMPORT.keys()]
-                logger.warning(
-                    "WARNING: search_alg type parameter missing, using 'variant_generator' as default. "
-                    f"These are possible values for the type parameter: {candiate_search_algs}."
-                )
-                search_alg = None
-            else:
-                search_alg_type = self.search_algorithm.search_alg_dict[TYPE]
-                search_alg = tune.create_searcher(
-                    search_alg_type, metric=metric, mode=mode, **self.search_algorithm.search_alg_dict
-                )
+        if self.search_algorithm is not None:
+            self.search_algorithm.set_random_state(random_seed)
+            sa_kwargs = self.search_algorithm.to_dict()
+            del sa_kwargs[TYPE]
+            sa_kwargs["mode"] = mode
+            sa_kwargs["metric"] = metric
+            search_alg = tune.create_searcher(self.search_algorithm.type, **sa_kwargs)
         else:
             search_alg = None
```

Comment:
Should we log a warning like we were before about the default search algorithm  we're going to be using?